src/springfield/entity.py

```diff
@@ -105,9 +107,11 @@ def jsonify(self):
         Return a dictionary suitable for JSON encoding.
         """
         data = {}
-        for key, val in self.__values__.iteritems():
```

Comment:
This whole problem was probably intended to be solved by "Personas". They don't exist in the code base yet but are planned and snippets of the code exist but have yet to be pull requested.

Example:

```
class UserEntity(Entity):
    id = fields.UuidField()
    name = fields.StringField()
    enabled = fields.BooleanField(default=True)
    created = fields.DateField(default=utcnow)

   class Create(Persona):
       # All fields are required on create
       id = required()
       name = required()
       enabled = required()
       created = required()

   class Update(Persona):
      # Only these fields can be set on update
       name = optional()
       enabled = optional()

   class List(Persona):
       id = required()
       name = required()
       enabled = required()
```

The personas are definitions for sparse entities that say exactly which fields are in that sparse entity. The persona extends the parent entity but only have the fields defined in that persona. They also allow specifying required fields which is expressly forbidden on the Entity itself because of the desire for sparse objects.

This means you can use `user = UserEntity.Create(name='Bob')` and do `user.validate()` and get an error if the additional `required` fields are not set. If defaults come in to play, all fields in the `Create` persona that have defaults could have them immediately. 

So maybe defaults should only apply to personas. However, I think there's a problem with that too. See the `UserEntity.List` persona. This would be used for the "get all users" case where you just want a quick index of users and to know which ones are `enabled`. If you didn't select `enabled` it's default value would be filled which puts us back to the same problem.

Maybe this can be solved by having `required`, `optional`, and `default` persona fields. So you could do:

```
   class Create(Persona):
       # All fields are required on create
       id = required()
       name = required()
       enabled = default()
       created = default()
```

This would make it so `created` and `enabled` are filled with default only when used for the `Create` persona and only if a value wasn't supplied. This would allow you to explicitly control when a default is populated. If you use `default` on a field that doesn't have a default value, it could be considered an error or `default()` could be considered equivalent to `required`. For this reason it might be better to have `required(allow_default=True)` instead of having `default()`.